### DISCUSSION TITLE
Error in createArrowFiles: some width(x) are less than 'n'

### DISCUSSION THREAD

**Discussion by tao-bioinfo** [GreenleafLab/ArchR#901] [Questions / Documentation] [answered]
My input file is `"outs_27_03/fragments.tsv.gz"` (together with tabix index) from `cellranger-atac count v2.0` of 10X genomic ATAC-seq.

My command is:
```
ArrowFiles = createArrowFiles(inputFiles = c("outs_27_03/fragments.tsv.gz"), sampleNames = c("27_03"), minTSS=2.5, addTileMat = TRUE, addGeneScoreMat = TRUE, geneAnnotation=geneAnnotation, genomeAnnotation=genomeAnnotation)
```

Full error log has been attached, and the most significant error what I see is:
```
2021-07-16 08:33:02 : ERROR Found in .tabixToTmp for (27_03 : 1 of 1) 
LogFile = ArchRLogs/ArchR-createArrows-c1ac24f1540f-Date-2021-07-16_Time-08-33-02.log

<simpleError in tile(x = x, n = n): some width(x) are less than 'n'>
```
[ArchR-createArrows-c1ac24f1540f-Date-2021-07-16_Time-08-33-02.log](https://github.com/GreenleafLab/ArchR/files/6826967/ArchR-createArrows-c1ac24f1540f-Date-2021-07-16_Time-08-33-02.log)

**Comment by rcorces** [comment, marked as answer]
Custom genomes are extremely difficult for us (the developers) to help troubleshoot. We have never worked outside of human and mouse and have really only enabled this feature to help end users with the hope that they can troubleshoot the problems that arise from their particular genomes. The number of users in human and mouse far outpaces the number of users in every other genome and we are already stretched very thin trying to maintain the package.

Presumably you have some really tiny scaffold? Presumably your error is occurring here:
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/CreateArrow.R#L1190

Try running that line in isolation where `chromSizes` is a `GenomicRanges` of your chromSizes and `nChunk` = 3
`unlist(GenomicRanges::tile(chromSizes, 3))`

**Reply by tao-bioinfo** [reply]
Yes, there are some very tiny scaffolds in my genome. However in such scaffolds there were no genes and also should be no reads mapping there. 

I'll try to rebuild a genome containing only main chromosomes and rerun the pipeline

**Reply by tao-bioinfo** [reply]
After rebuilding a genome with only main chromosomes, this error has been fixed.
